src/libraries/networking/webrtc/WebRTCDataChannel.js

```diff
@@ -0,0 +1,303 @@
+//
+//  WebRTCDataChannel.js
+//
+//  WebRTC connection and data channel to the domain server or an assignment client. (Each uses their own WebRTC connection.)
+//
+//  Created by David Rowe on 21 May 2021.
+//  Copyright 2021 Vircadia contributors.
+//
+//  Distributed under the Apache License", Version 2.0.
+//  See the accompanying file LICENSE or http://www.apache.org/licenses/LICENSE-2.0.html
+//
+
+/* globals RTCPeerConnection */
+
+import NodeType from "../NodeType.js";
+import WebRTCSignalingChannel from "./WebRTCSignalingChannel.js";
+
+// WebRTCDataChannel used for Vircadia protocol communications with a domain server or assignment client. Uses a
+// WebRTCSignalingChannel in the process of establishing the WebRTC connection.
+// The API is similar to the WebRTCSignalingChannel and WebSocket APIs.
+class WebRTCDataChannel {
+
+    /* eslint-disable no-magic-numbers */
+
+    // ReadyState values.
+    static CONNECTING = 0;
+    static OPEN = 1;
+    static CLOSING = 2;
+    static CLOSED = 3;
+
+    /* eslint-enable no-magic-numbers */
+
+    static #CONFIGURATION = {
+        iceServers: [{ urls: "stun:ice.vircadia.com:7337" }]
```

Comment:
This should be retrieved from a configuration file at some point. Same should apply to any other networking constants.

Reply:
The C++ equivalent is `ICE_SERVER_DEFAULT_HOSTNAME` in NetworkingConstants.h, though it can be overridden by a command line parameter when starting the domain server - https://docs.vircadia.com/developer/domain-server/command-line-parameters.html.
It may perhaps be necessary for Interface to get the ICE server address from the metaverse server for the chosen metaverse.

Reply:
Makes sense to me.